pkg/osquery/tables/gsettings/gsettings.go

```diff
@@ -0,0 +1,184 @@
+// +build !windows
+
+package gsettings
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	osquery "github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const gsettingsPath = "/usr/bin/gsettings"
+
+const allowedCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-."
+
+type gsettingsExecer func(ctx context.Context, username string, buf *bytes.Buffer) error
+
+type GsettingsValues struct {
+	client   *osquery.ExtensionManagerClient
+	logger   log.Logger
+	getBytes gsettingsExecer
+}
+
+// Settings returns a table plugin for querying setting values from the
+// gsettings command.
+func Settings(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("schema"),
+		table.TextColumn("key"),
+		table.TextColumn("value"),
+		table.TextColumn("user"),
+	}
+
+	t := &GsettingsValues{
+		client:   client,
+		logger:   logger,
+		getBytes: execGsettings,
+	}
+
+	return table.NewPlugin("kolide_gsettings", columns, t.generate)
+}
+
+func (t *GsettingsValues) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	users := tablehelpers.GetConstraints(queryContext, "user", tablehelpers.WithAllowedCharacters(allowedCharacters))
+	if len(users) < 1 {
+		return results, errors.New("kolide_gsettings requires at least one user name to be specified")
+	}
+	for _, username := range users {
+		var output bytes.Buffer
+
+		err := t.getBytes(ctx, username, &output)
+		if err != nil {
+			level.Info(t.logger).Log(
+				"msg", "error getting bytes for user",
+				"username", username,
+				"err", err,
+			)
+			continue
+		}
+
+		user_results := t.parse(&output)
+		for _, r := range user_results {
+			r["user"] = username
+			results = append(results, r)
+		}
```

Comment:
(nit) Feels weird to loop twice. Consider moving the username _into_ the row creation inside `parse`